conftest.py

```diff
@@ -165,9 +166,9 @@ def pytest_sessionfinish(
                     raise
             else:
                 break
-
-    # kill all spawned processes
-    kill_all_test_spawned_processes()
+    else:
+        # kill all spawned processes
+        kill_all_test_spawned_processes()
```

Comment:
Are we sure we want to leave extra processes hanging around when a test fails?

Reply:
yeah, that was my thought (kill on success or failure to avoid all zombies). i'm leaning toward your solution now. resolving.